receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,306 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+type logsReceiver struct {
+	log         *zap.Logger
+	cfg         *Config
+	client      *internal.MongoDBAtlasClient
+	consumer    consumer.Logs
+	stopperChan chan struct{}
+	wg          sync.WaitGroup
+}
+
+type ProjectContext struct {
+	Project   mongodbatlas.Project
+	orgName   string
+	projectID string
+	start     time.Time
+	end       time.Time
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collectionInterval = time.Minute * 5
+
+func newMongoDBAtlasLogsReceiver(settings component.ReceiverCreateSettings, cfg *Config, consumer consumer.Logs) (*logsReceiver, error) {
+	client, err := internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger)
+	if err != nil {
+		return nil, err
+	}
+	return &logsReceiver{
+		log:         settings.Logger,
+		cfg:         cfg,
+		client:      client,
+		stopperChan: make(chan struct{}),
+		consumer:    consumer}, nil
+}
+
+// Log receiver logic
+func (s *logsReceiver) Start(ctx context.Context, host component.Host) error {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		start := time.Now()
```

Comment:
Should this be initialized based on the collection interval? (ie. time.Now() - 5 minutes)

Otherwise, start and end are essentially the same for the first collection.

Reply:
That is true, but since it goes through the select statement prior to going to the `collect` method it should wait the collection interval before collection. Granted, I think when you start a receiver you probably shouldn't wait first and then collect logs, so I will switch that around as well.

Reply:
Yeah, that makes sense. "Immediate" feedback is important for anyone working on a config.